src/index.ts

```diff
@@ -1,17 +1,47 @@
 import * as core from "@actions/core";
 import { readConfig } from "./config";
 import { apiClient } from "./client/apiClient";
+import { isSuccessful } from "./run/status";
+import { startRun } from "./run/start";
+import { FinishedRun, waitForRunEnd } from "./run/ongoing";
 
 async function run(): Promise<void> {
   try {
     const config = readConfig();
     const client = apiClient(config.api);
-    const simulations = await client.getSimulations(); // TODO RND-6 Start simulation instead
-    core.setOutput("result", `Found {} ${simulations.length} simulations.`);
+
+    await client.checkCloudCompatibility();
+
+    const startedRun = await startRun(client, config);
+    core.setOutput("run_id", startedRun.runId);
+    core.setOutput("reports_url", startedRun.reportsUrl);
+    core.setOutput("runs_url", startedRun.runsUrl);
+
+    const finishedRun = await waitForRunEnd(client, startedRun);
+    logResult(finishedRun);
+
+    // TODO RND-7 Metrics summary and XML report
+
+    core.setOutput("runs_status_code", finishedRun.statusCode);
+    core.setOutput("runs_status_name", finishedRun.statusName);
+    core.setOutput("run_assertions", JSON.stringify(finishedRun.assertions));
```

Comment:
Only string for `setOutput`?
Can be interesting for users to be able to manipulate the value directly in their actions, I think

Reply:
Outputs are only strings, yes. Though the doc for `core.setOuput` does say "Non-string values will be converted to a string via JSON.stringify", so I can avoid doing the conversion myself.